tests/utils/site-builder.js

```diff
@@ -130,11 +130,21 @@ const createSiteBuilder = ({ siteName }) => {
       })
       return builder
     },
+    withoutFile: ({ path: filePath }) => {
+      const dest = path.join(directory, filePath)
+      tasks.push(async () => {
+        await fs.rmFileAsync(dest)
+      })
+      return builder
+    },
     buildAsync: async () => {
       for (const task of tasks) {
         // eslint-disable-next-line no-await-in-loop
         await task()
       }
+
+      tasks = []
```

Comment:
By emptying the tasks array at the end of `buildAsync`, we're able to run new batches of tasks. For example, we can use `builder.withFunction` to create/update a function after the site has been built and served, which is useful when testing `netlify dev`.